fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RecordConstructorValue.java

```diff
@@ -286,12 +286,7 @@ public Accumulator createAccumulator(final @Nonnull TypeRepository typeRepositor
             @Override
             public void accumulate(@Nullable final Object currentObject) {
                 if (childAccumulators == null) {
-                    final ImmutableList.Builder<Accumulator> childAccumulatorsBuilder = ImmutableList.builder();
-                    for (final var child : getChildren()) {
-                        Verify.verify(child instanceof AggregateValue);
-                        childAccumulatorsBuilder.add(((AggregateValue)child).createAccumulator(typeRepository));
-                    }
-                    childAccumulators = childAccumulatorsBuilder.build();
+                    childAccumulators = buildAccumulators();
```

Comment:
Can this just be done in the "constructor," that is, make `childAccumulators` final? I believe `buildAccumulators` works in the parent class, if `typeRespository` is made an argument.

Reply:
done.